crates/bevy_ecs/macros/src/component.rs

```diff
@@ -17,6 +17,21 @@ pub fn derive_resource(input: TokenStream) -> TokenStream {
     let mut ast = parse_macro_input!(input as DeriveInput);
     let bevy_ecs_path: Path = crate::bevy_ecs_path();
 
+    // We want to raise a compile time error when the generic lifetimes
+    // are not bound to 'static lifetime
+    let non_static_lifetime_error: Option<syn::Error> = ast
```

Comment:
I think this type can be inferred:

```suggestion
    let non_static_lifetime_error = ast
```